backend/homework/7/src/main/java/sanskar/service/VehicleServiceFactory1.java

```diff
@@ -0,0 +1,63 @@
+package sanskar.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import sanskar.inventory.InventoryStore;
+import sanskar.model.Vehicle;
+
+import java.util.Comparator;
+
+@Service
+public class VehicleServiceFactory1 extends VehicleService {
+
+    @Autowired
+    private TyreService tyreService;
+
+    @Autowired
+    private SpeakerService speakerService;
+
+    private final InventoryStore inventoryStore;
+
+    @Autowired
+    public VehicleServiceFactory1(InventoryStore inventoryStore) {
+        this.inventoryStore = inventoryStore;
+    }
+
+    @Override
+    public void generateVehicles() {
```

Comment:
Can be moved to utils and made more cleaner in presentation. Use switch cases, indent properly.